ddl/ddl.go

```diff
@@ -665,3 +665,13 @@ func (d *ddl) GetHook() Callback {
 
 	return d.mu.hook
 }
+
+// RecoverInfo contains information needed by DDL.RecoverTable.
+type RecoverInfo struct {
+	SchemaID      int64
+	TableInfo     *model.TableInfo
+	DropJobID     int64
+	SnapshotTS    uint64
+	CurAutoIncID  int64
+	CurAutoRandID int64
+}
```

Comment:
meta like `xxxInfo` should be defined in the parser... strange here.
how about just append it in parameters.

Reply:
The `RecoverInfo` is not a meta-structure. Here it is only used as a "composite" parameter in `RecoverTable()` to reduce the number of parameters. BTW, I think passing pointers is more efficient than copying multiple arguments.